src/CoreWCF.Primitives/src/CoreWCF/Channels/ChannelBuilder.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime;
+using CoreWCF.Configuration;
+using CoreWCF.Dispatcher;
+namespace CoreWCF.Channels
+{ 
+    class ChannelBuilder
+    {
+        CustomBinding binding;
+        BindingContext context;
+        BindingParameterCollection bindingParameters;
+        Uri listenUri;
+
+        public ChannelBuilder(BindingContext context, bool addChannelDemuxerIfRequired)
+        {
+            this.context = context;
+            if (addChannelDemuxerIfRequired)
+            {
+                throw new NotImplementedException();
+                //TODO next phase
```

Comment:
NotImplementedException is meant for when you have a base definition such as an interface or abstract class and your concrete implementation doesn't implement that method. PlatformNotSupportedException is a better fit here.